openquakeplatform/openquakeplatform/ript/static/ript/js/vulnerability_new.js

```diff
@@ -0,0 +1,525 @@
+                }
+            }
+
+            var imlArray = discreteImlHeader[k-1].imlValue;
+            imlArray = imlArray.split(" ");
+            for (var i = 0; i < imlArray.length; i++) {
+                // prob stuff
+                var temp = tempProb[i+1].join(" ");
+                probabilities += '\t\t\t<probabilities lr="'+imlArray[i]+'">'+temp+'</probabilities>\n';
+            }
+        }
+
+        // Opening vfs tag
+        vfs = '\t\t<vulnerabilityFunction id="'+idObj[k]+'" dist="'+discreteImlHeader[k-1].distributionType+'">\n';
+
+        // Create the imls, meanLRs and covLRs tags
+        var imlsTag;
+        if (discreteImlHeader[k-1].functionType == 'probability mass function') {
+            imlsTag = '\t\t\t<imls imt="'+discreteImlHeader[k-1].imtValue+'">'+tempProb[0].join(" ")+'</imls>\n';
+            vfs += imlsTag;
+        } else if (discreteImlHeader[k-1].functionType == 'discrete') {
+            // Manage SA cases
+            if (discreteImlHeader[k-1].imtValue == 'sa') {
+                imlsTag = '\t\t\t<imls imt="'+(discreteImlHeader[k-1].imtValue).toUpperCase()+' ('+discreteImlHeader[k-1].saValue+')">';
+            } else {
+                imlsTag = '\t\t\t<imls imt="'+(discreteImlHeader[k-1].imtValue).toUpperCase()+'">';
+            }
+
+            meanLRsTag = '\t\t\t<meanLRs>';
+            covLRsTag = '\t\t\t<covLRs>';
+            for (var i = 0; i < dataVF[k].length; i++) {
+                if (i !== (dataVF[k].length -1)) {
+                    imlsTag += dataVF[k][i][0]+' ';
+                    meanLRsTag += dataVF[k][i][1]+' ';
+                    covLRsTag += dataVF[k][i][2]+' ';
+                } else {
+                    // Avoid trailing whitespace
+                    imlsTag += dataVF[k][i][0];
+                    meanLRsTag += dataVF[k][i][1];
+                    covLRsTag += dataVF[k][i][2];
+                }
+            }
+
+            // Closing tags
+            imlsTag += '</imls>\n';
+            meanLRsTag += '</meanLRs>\n';
+            covLRsTag += '</covLRs>\n';
+
+            // Put all the tags together
+            vfs += imlsTag;
+            vfs += meanLRsTag;
+            vfs += covLRsTag;
+        }
+
+        if (discreteImlHeader[k-1].functionType == 'probability mass function') {
+            vfs += probabilities;
+        }
+
+        // Closing vfs tags
+        vfs += '\t\t</vulnerabilityFunction>\n';
+        vulnFunction += vfs;
+    }
+
+    // Create a NRML element
+    NRML =
+        '<?xml version="1.0" encoding="UTF-8"?>\n' +
+        '<nrml xmlns="http://openquake.org/xmlns/nrml/0.5">\n' +
+            '\t<vulnerabilityModel id="'+functionId+'" assetCategory="'+assetCategory+'" lossCategory="'+lossCategory+'">\n' +
+                '\t\t<description>'+functionDescription+'</description>\n' +
+                vulnFunction +
+            '\t</vulnerabilityModel>\n' +
+        '</nrml>';
+
+    validateAndDisplayNRML(NRML, 'VF')
```

Comment:
I think `VF` should become `vf`, but perhaps the whole `vulnerability_new.js` file will be removed?